commit b00839ca4cca8aa9641c121c848a553d6220ce70
Author: Justin Stitt <justinstitt@google.com>
Date:   Wed Sep 13 17:13:04 2023 +0000

    ipmi: refactor deprecated strncpy
    
    `strncpy` is deprecated for use on NUL-terminated destination strings [1].
    
    In this case, strncpy is being used specifically for its NUL-padding
    behavior (and has been commented as such). Moreover, the destination
    string is not required to be NUL-terminated [2].
    
    We can use a more robust and less ambiguous interface in
    `memcpy_and_pad` which makes the code more readable and even eliminates
    the need for that comment.
    
    Let's also use `strnlen` instead of `strlen()` with an upper-bounds
    check as this is intrinsically a part of `strnlen`.
    
    Also included in this patch is a simple 1:1 change of `strncpy` to
    `strscpy` for ipmi_ssif.c. If NUL-padding is wanted here as well then we
    should opt again for `strscpy_pad`.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://lore.kernel.org/all/ZQEADYBl0uZ1nX60@mail.minyard.net/ [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Cc: Kees Cook <keescook@chromium.org>
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Message-Id: <20230913-strncpy-drivers-char-ipmi-ipmi-v2-1-e3bc0f6e599f@google.com>
    Signed-off-by: Corey Minyard <minyard@acm.org>

diff --git a/drivers/char/ipmi/ipmi_ssif.c b/drivers/char/ipmi/ipmi_ssif.c
index df8dd50b4cbe..1f7600c361e6 100644
--- a/drivers/char/ipmi/ipmi_ssif.c
+++ b/drivers/char/ipmi/ipmi_ssif.c
@@ -1945,7 +1945,7 @@ static int new_ssif_client(int addr, char *adapter_name,
 		}
 	}
 
-	strncpy(addr_info->binfo.type, DEVICE_NAME,
+	strscpy(addr_info->binfo.type, DEVICE_NAME,
 		sizeof(addr_info->binfo.type));
 	addr_info->binfo.addr = addr;
 	addr_info->binfo.platform_data = addr_info;